File: src/main/java/net/namekdev/graphql2elm/WebStarter.java
package net.namekdev.graphql2elm;

import org.teavm.jso.browser.*;
import org.teavm.jso.dom.events.Event;
import org.teavm.jso.dom.events.EventListener;
import org.teavm.jso.dom.html.*;

import java.util.Locale;

import static org.antlr.v4.runtime.atn.ATNDeserializer.SERIALIZED_VERSION;


public class WebStarter {

    public static void main(String[] args) {
        HTMLDocument document = Window.current().getDocument();

        HTMLElement preEl = document.createElement("pre");
        preEl.setInnerHTML("hey just a test");
        document.getBody().appendChild(preEl);
    }
}

File: src/main/kotlin/net/namekdev/graphql2elm/WebStarter.kt
package net.namekdev.graphql2elm

import org.teavm.jso.browser.*
import org.teavm.jso.dom.events.Event
import org.teavm.jso.dom.events.EventListener
import org.teavm.jso.dom.html.*

fun main(args: Array<String>) {
    val document = Window.current().document

    val preEl = document.createElement("pre")
    preEl.innerHTML = "hey just a test"
    document.body.appendChild(preEl)

    val str1 = """
    query {
      currentUser {
        balances {
          otherUser {
            id
            name
          }
          value
          iHaveMore
          sharedPaymentCount
          transferCount
          unseenUpdateCount
        }
      }
    }
    """

    preEl.innerHTML = generateElmCode(query = str1, schema = queryForSchema())
}